src/plugins/displayLayout/components/DisplayLayout.vue

```diff
@@ -99,86 +102,96 @@
         data() {
             return {
                 gridSize: [],
-                frameItems: [],
-                frames: [],
-                frameStyles: [],
-                rawPositions: {},
+                layoutItems: [],
                 drilledIn: undefined
             }
         },          
         inject: ['openmct'],
         props: ['domainObject'],
         components: {
-            LayoutFrame
+            LayoutItem
         },
         created: function () {
             this.newDomainObject = this.domainObject;
             this.gridSize = this.newDomainObject.layoutGrid ||  DEFAULT_GRID_SIZE;
-            this.composition = this.openmct.composition.get(this.newDomainObject);
-            this.Listeners = [];
-            let panels = (((this.newDomainObject.configuration || {}).layout || {}).panels || {});
 
-            if (this.composition !== undefined) {
-                this.composition.load().then((composition) => {
-                    composition.forEach(function (domainObject) {
-                        this.readLayoutConfiguration(domainObject, panels);
-                        this.makeFrameItem(domainObject, false);
-                    }.bind(this));
-                    this.composition.on('add', this.onAddComposition);
-                    this.composition.on('remove', this.onRemoveComposition);
-                });
-            }
+            // Read layout configuration
+            this.getPanels();
+            this.getAlphanumerics();
 
             this.unlisten = this.openmct.objects.observe(this.newDomainObject, '*', function (obj) {
                 this.newDomainObject = JSON.parse(JSON.stringify(obj));
                 this.gridSize = this.newDomainObject.layoutGrid || DEFAULT_GRID_SIZE;;
             }.bind(this));
         },
         methods: {
-            readLayoutConfiguration(domainObject, panels) {
-                let id = this.openmct.objects.makeKeyString(domainObject.identifier);
-                this.rawPositions[id] = {
-                    position: panels[id].position || DEFAULT_POSITION,
-                    dimensions: panels[id].dimensions || this.defaultDimensions()
+            getPanels() {
+                let panels = this.newDomainObject.configuration.panels;
+
+                for (const id in panels) {
```

Comment:
to use `for ... in` you need to check each id using `hasOwnProperty`.  But thanks to modern JS, you should use `for ... of` with Object.entries in order to iterate over key/value pairs, e.g. `for (const [id, panel] of Object.entries(panels)) { ... }`.

Reply:
getPanels() is removed and so this code doesn't apply.